### DISCUSSION TITLE
OpENer port to STM32F7 and lwIP

### DISCUSSION THREAD

**Discussion by FrikBaden** [EIPStackGroup/OpENer#377] [Q&A]
I am trying to port OpENer to the STM32F7 with lwIP.  Have anyone done this before successfully and will it be possible to get a copy of the ported software? I cannot see anything on the OpENer github.

**Comment by MartinMelikMerkumians** [comment]
Didn't try it and don't have a port for STM32, sorry.
Perhaps someone else tried this?

**Reply by MartinMelikMerkumians** [reply]
Seems PeterChristen (#346 contributer) tries to port to STM32

**Comment by mmprestine** [comment]
I have ported to STM32 and it is not much work.  The porting really has nothing to do with OpEner if done properly, it is the project source and asking for that might be a bit much.

**Comment by Aaron-Levan** [comment]
This is a late reply, but I'm sure there are others trying to do this as well.
I have ported OpENer to a few different MCU's using FreeRTOS and lwIP. 
I will **try** to make some time to dig it back up, clean it up and post some guidance.

**Comment by alexbussiere** [comment]
We are looking at porting this to the ESP32 with FreeTOS and lwIP and any info people have of their implementations / modifications for embedded environments would be really appreciated!

**Reply by MartinMelikMerkumians** [reply]
https://github.com/EIPStackGroup/OpENer/pull/457

What you are looking for should he in the current pull request

**Reply by alexbussiere** [reply]
What Ethernet/IP Scanner did you have that working with? I am trying it with an Allen Bradley PLC and keep running into an error that the output size is incorrect

**Reply by alexbussiere** [reply]
Never mind, I didn't have OPENER_CONSUMED_DATA_HAS_RUN_IDLE_HEADER defined, that seemed to do the trick and now its connected
